### PR TITLE
refactor: add multiplier to TokenSenderCaller

### PR DESCRIPTION
**Pull request by ramenforbreakfast**
One thing to note, is our old `TokenSender`, used a smaller precision denominator of 10000, I decided we should just keep it simple and the same as our fee denominators. I would also prefer if we just use the existing naming of PERCENT_DENOMINATOR, but not sure what we would prefer for the naming.

### CODE REVIEW

File: packages/prepo-shared-contracts/contracts/TokenSenderCaller.sol [hunk 1]
```diff
@@ -5,13 +5,29 @@ import "./interfaces/ITokenSenderCaller.sol";
 import "./interfaces/ITokenSender.sol";
 
 contract TokenSenderCaller is ITokenSenderCaller {
+  uint256 internal _amountMultiplier;
   ITokenSender internal _tokenSender;
 
+  uint256 public constant override MULTIPLIER_DENOMINATOR = 1000000;
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/contracts/TokenSenderCaller.sol:11]:
```suggestion
  uint256 public constant override MULTIPLIER_UNIT = 1000000;
```
?

File: packages/prepo-shared-contracts/contracts/interfaces/ITokenSenderCaller.sol [hunk 2]
```diff
@@ -6,7 +6,15 @@ import "./ITokenSender.sol";
 interface ITokenSenderCaller {
   event TokenSenderChange(address sender);
 
+  event AmountMultiplierChange(uint256 multiplier);
+
   function setTokenSender(ITokenSender tokenSender) external;
 
-  function getTokenSender() external returns (ITokenSender);
+  function setAmountMultiplier(uint256 amountMultiplier) external;
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/contracts/interfaces/ITokenSenderCaller.sol:13]:
```suggestion
  function setAmountMultiplier(uint256 multiplier) external;
```

File: packages/prepo-shared-contracts/contracts/TokenSenderCaller.sol [hunk 3]
```diff
@@ -5,13 +5,29 @@ import "./interfaces/ITokenSenderCaller.sol";
...
+
+  function setAmountMultiplier(uint256 amountMultiplier)
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/contracts/TokenSenderCaller.sol:13]:
```suggestion
  function setAmountMultiplier(uint256 multiplier)
```

File: packages/prepo-shared-contracts/contracts/interfaces/ITokenSenderCaller.sol [hunk 4]
```diff
@@ -6,7 +6,15 @@ import "./ITokenSender.sol";
...
+
+  function getTokenSender() external view returns (ITokenSender);
+
+  function getAmountMultiplier() external view returns (uint256);
+
+  function MULTIPLIER_DENOMINATOR() external view returns (uint256);
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/contracts/interfaces/ITokenSenderCaller.sol:19]:
```suggestion
  function MULTIPLIER_UNIT() external view returns (uint256);
```

File: packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts [hunk 5]
```diff
@@ -19,8 +19,53 @@ describe('=> TokenSenderCaller', () => {
   })
 
   describe('initial state', () => {
+    it('does not set amount multiplier', async () => {
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts:22]:
```suggestion
    it('doesn't set amount multiplier', async () => {
```

File: packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts [hunk 6]
```diff
@@ -19,8 +19,53 @@ describe('=> TokenSenderCaller', () => {
...
+      expect(await tokenSenderCaller.getAmountMultiplier()).eq(0)
+    })
+
     it('does not set TokenSender', async () => {
-      expect(await tokenSenderCaller.getTokenSender()).to.eq(ZERO_ADDRESS)
+      expect(await tokenSenderCaller.getTokenSender()).eq(ZERO_ADDRESS)
+    })
+
+    it('sets MULTIPLIER_DENOMINATOR constant', async () => {
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts:30]:
```suggestion
    it('sets MULTIPLIER_UNIT constant', async () => {
```

File: packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts [hunk 7]
```diff
@@ -19,8 +19,53 @@ describe('=> TokenSenderCaller', () => {
...
+      expect(await tokenSenderCaller.MULTIPLIER_DENOMINATOR()).eq(MULTIPLIER_DENOMINATOR)
+    })
+  })
+
+  describe('# setAmountMultiplier', () => {
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts:35]: add tests for setting = and > `MULTIPLIER_UNIT`

File: packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts [hunk 8]
```diff
@@ -19,8 +19,53 @@ describe('=> TokenSenderCaller', () => {
...
+    it('sets to non-zero', async () => {
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/test/TokenSenderCaller.test.ts:36]:
```suggestion
    it('sets to non-zero value < MULTIPLIER_UNIT', async () => {
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: packages/prepo-shared-contracts/contracts/TokenSenderCaller.sol (hunk 3 above)
Comment (ramenforbreakfast) [reply packages/prepo-shared-contracts/contracts/TokenSenderCaller.sol:13]: Im not going to do this change, and will hold off on what is decided in 1188

**Pull request merged**